languages/csharp/exercises/concept/interfaces/.meta/Example.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+
+public interface IRemoteControlCar
+{
+    void Drive();
+    decimal DistanceTravelled { get; }
+}
+
+public class ProductionRemoteControlCar : IRemoteControlCar, IComparable<ProductionRemoteControlCar>
+{
+    public decimal DistanceTravelled { get; private set; }
+    public int NumberOfVictories { get; set; }
+
+    public void Drive()
+    {
+        DistanceTravelled += 10;
+    }
+
+    public int CompareTo(ProductionRemoteControlCar other)
+    {
+        if (ReferenceEquals(this, other)) return 0;
+        if (ReferenceEquals(null, other)) return 1;
+        return NumberOfVictories.CompareTo(other.NumberOfVictories);
+    }
+}
+
+public class ExperimentalRemoteControlCar : IRemoteControlCar
+{
+    public decimal DistanceTravelled { get; private set; }
+
+    public void Drive()
+    {
+        DistanceTravelled += 20;
+    }
+}
+
+public static class TestTrack
+{
+    public static decimal Race(IRemoteControlCar car)
+    {
+        car.Drive();
+        return car.DistanceTravelled;
+    }
+
+    public static List<ProductionRemoteControlCar> GetRankedCars(ProductionRemoteControlCar prc1,
+        ProductionRemoteControlCar prc2)
+    {
+        var rankings = new List<ProductionRemoteControlCar>{prc1, prc2};
+        rankings.Sort();
+        return rankings;
+    }
```

Comment:
As there are only two cars involved, there is a possibility that students hard-code the logic using an `if-statement`. What if we change the method a bit to take a list of production cars and expect it to be sorted after the call? Something like this:

```suggestion
    public static void RankCars(List<ProductionRemoteControlCar> unrankedCars)
    {
        unrankedCars.Sort();
    }
```

As an alternative, we could just have the student implement `IComparable<T>` and do the sorting ourselves in the test (see a previous comment I made).

Reply:
@mikedamay This comment is marked as resolved, but I don't see any change? Was that intentional?